lib/block_account.go

```diff
@@ -159,3 +160,97 @@ func (b *BlockAccount) Withdraw(fund Amount, checkpoint string) error {
 	}
 	return nil
 }
+
+// BlockAccountCheckpoint is the one-and-one model of account and checkpoint in
+// block. the storage should support,
+//  * find by `Address`:
+// 	- key: "`Address`-`Checkpoint`": value: `ID` of BlockAccountCheckpoint
+//  * get list by created order:
+//
+// models
+//  * 'address' and 'checkpoint'
+// 	- 'bac-<BlockAccountCheckpoint.Address>-<BlockAccountCheckpoint.Checkpoint>': `BlockAccountCheckpoint`
+type BlockAccountCheckpoint struct {
+	Checkpoint string
```

Comment:
I think, if checkpoint type is the struct like this,
```
type Checkpoint struct {
  Source string
  Target string
}
```
then it would be good for improving performance. But it is quite big changes so, we can handle it next issue #98.

Reply:
@Charleslee522 We can make it in new issue :)